app/assets/javascripts/bed_verification.js

```diff
@@ -94,6 +96,8 @@
         if (!this) {
           $('#bed_list li[data-bed="' + bed_id + '"]').addClass('bad_bed list-group-item-danger')
           bad_beds.push(bed_id)
+        } else {
+          clearFlagFromBed(bed_id)
```

Comment:
This is a bit complex, What is 'this' for? From jQuery help, they say 'this' is the element, however we are iterating a normal array, not DOM, so I assume 'this' is the element, but then it will never be undefined, son line 97 would never reach?

Reply:
The method name is flagBeds, but it is removing the flag from a bed, and only adding it if this is not defined, but I dont see when that happens.

Reply:
I think this method is never doing the if (!this) part of the condition. Example: 

```$.each([1,2,3], function(a) { console.log("this => "+this)})```

Reply:
It does reach the !this statement and set the flag to error so it does something, but I'm not a JS expert so had assumed it was short for if bed not valid

Reply:
Thanks for checking it with me, I didnt know this behaviour of $.each when using hashes. So when: 
```
$.each({a: 1, b: 3}, function(bed_id) { console.log(this) })
```
The output is 1 and 3 (the value of the element)

Reply:
And in this case it seems the hash is ```{'bed_1': true, 'bed_2': false)```, so here ```this``` is a boolean

Reply:
Fixed